code-reviews.md

```diff
@@ -0,0 +1,80 @@
+However, it is important to keep in mind that what is under review must be understandable in its software context.
+Some complex refactorings or features can not be split in small changes as they would lose software design information.
+The reviewer would then fail to comment on design choices and be limited to a simple technical review.
+
+When opening a pull request \(PR\) or merge request \(MR\) for a review, we recommend to assess the granularity of the review to keep visible the algorithm and design of the feature.
+
+## Best practices for reviewing codes
+
+The challenge of a code review is to position oneself in the face of another member's work production.
+It is not a question of sanctioning his/her work, but of assessing it; issuing an opinion.
+
+Here are 5 good practices to adopt that may be beneficial for a code review.
+
+1. Engage conversation for clarification 💬
+
+The point here is to question what has been produced.
+It is not because the reviewer has difficulty understanding the code that the code quality is bad.
+Code reviews is an exercise that enhances the value of the work produced in a team spirit.
+We recommend to asking the developer why he/she did it rather than being prescriptive and tell developer to code in a specific way.
+
+> I am not sure to understand. What was your point when doing this
+
+Also, to discuss about the choices made by the developer, we encourage to suggest alternatives that could optimize what has been done.
+
+> Doing it like this could solve the problem in only one step. What do you think of it ? (WDYT ?)
+
+2. Have a pragmatic state of mind ✔️
+
+It is important to keep in mind that what is provided is functional.
+External parameters can force the developer not to provide optimal code.
+He/she is not omniscient and learns from teamwork too.
+Having a functional code is the paramount issue (considering that good practices of coding are respected).
+
+3. Review as often as possible 🕙
+
+Code reviews will be relevant and beneficial if the exercise is frequent within the team.
+It is not the business of a few members but of each one of them because such exercises contribute to the good integration and involvement of everyone.
+
+4. Focus on the main 🧠
+
+A code review should not be overthought.
+The reviewer should focus on the algorithm and the code.
+With a problem comes a solution.
+It must be possible to be synthetic and have a global vision of what is in question.
+
+5. Communicate positively and with consideration 😃
+
+Finally, the most important is the way to communicate during a review.
+We want to keep a warm and team spirit in the perspective of celebrating and encouraging progress in a joyful and motivating mood.
+In addition to ensuring that positive and enthusiastic comments are always written, we support the use of gifs and emojis in messages and \(PR)\ or \(MR\) conversations 😉

```

Comment:
```suggestion
In addition to ensuring that positive and enthusiastic comments are always written, we encourage? the use of gifs and emojis in messages and \(PR)\ or \(MR\) conversations 😉

```

Reply:
I just wanted to use an other word as I wrote "encouraging" in the previous sentence.

Reply:
It's much better !! 
[![Alt text](https://media.giphy.com/media/YOZ7IdqmLJTZ93LUEX/giphy.gif)

maybe as there is a debat :  In addition to ensuring that positive and enthusiastic comments are always written, don't hesitate on using gifs and emojis in messages and \(PR)\ or \(MR\) conversations 😉